docs/configuration/secure.md

```diff
@@ -0,0 +1,571 @@
+    + [Redirect all unencrypted traffic to HTTPS](#redirect-all-unencrypted-traffic-to-https)
+    + [Proper SSL configuration](#proper-ssl-configuration)
+    + [Serve security related Headers](#serve-security-related-headers)
+    + [Enable OCSP stapling (TLS Certificate Status Request)](#enable-ocsp-stapling--tls-certificate-status-request-)
+    + [Enable HTTP Strict Transport Security](#enable-http-strict-transport-security)
+    + [Genrate the ILIAS ```ilClientId``` cookie with ```secure``` attribute](#genrate-the-ilias----ilclientid----cookie-with----secure----attribute)
+  * [Suppress server signature and PHP version information](#suppress-server-signature-and-php-version-information)
+  * [deny access or restrict to several files or locations](#deny-access-or-restrict-to-several-files-or-locations)
+    + [ILIAS setup](#ilias-setup)
+    + [Prevent blacklisted files of beeing served by the webserver](#prevent-blacklisted-files-of-beeing-served-by-the-webserver)
+    + [Prevent execution of PHP-Code in data-directory](#prevent-execution-of-php-code-in-data-directory)
+    + [Deny Access to local Git-Directory](#deny-access-to-local-git-directory)
+  * [Integrity check of ILIAS code in docroot](#integrity-check-of-ilias-code-in-docroot)
+  * [Use WebAccessChecker](#use-webaccesschecker)
+  * [Use secure passwords](#use-secure-passwords)
+  * [Report security issues](#report-security-issues)
+
+# Hardening and Security Guidance
+
+This guideline is going to show you some best practice examples how to improve the security of your ILIAS installation, of the webserver and it's associated components.
+The ILIAS e.V. requested a documentation for a "Secure ILIAS" in march 2019.
+The section "Hardening and Security Guidance" should be removed and the security related instructions have to be mantained in a own document.
+
+## Firewall
+
+Block all traffic by default and explicitly allow only specific traffic to port 443/TCP for HTTPS secured traffic.
+For "quality of life", it is recommend to also permit 80/TCP for a [redirect to HTTPS](#redirect-all-unencrypted-traffic-to-https-1).
+
+## File Access Rights
+
+If you're an experienced admin you MAY want to use more restrictive file access rights that we RECOMMEND in this document. To make it impossible for an attacker to modify PHP files if he gains control over the web server processes, those files SHOULD be owned by ```root``` wherever possible.
+
+The only files and directories that MUST be owned/writeable by the web user are:
+
+  * ilias.ini.php
+  * data/
+  * ILIAS data dir outside of the webservers docroot
+
+All the other files and directories should be owned by ```root```, but readable by the web user (e.g. 644/755).
+
+example of the suggested configuration:
+```
+git clone --single-branch -b release_5-4 https://github.com/ILIAS-eLearning/ILIAS.git \ /var/www/ilias/.
+
+mkdir -p /var/www/ilias;
+mkdir /var/www/ilias_data;
+mkdir /var/www/ilias_log;
```

Comment:
Same here: ``/var/log/ilias`` seems to me the FHS standard place instead.

Reply:
I agree with that too. This is the path a system administrator unfamiliar with ILIAS would look for log files.

Reply:
I'ved change the path an fix an issue related to open_basedir.